### PR TITLE
[DOCS] Add C-C++ interoperability example

### PR DESCRIPTION
**Pull request by Krishna-13-cyber**
This PR adds the tutorial for C-C++ interoperability in the documentation.
This is on top of #105

### CODE REVIEW

File: docs/tutorials.rst [hunk 1]
```diff
@@ -181,4 +181,123 @@ function is called, which is `callme` function in this case.
...
+   
+C:
+===
+
+Include **p3-ex4-lib.h**, which contains the declarations for the functions used
+in this code. The detailed summary of header comes in the latter part.
+
+
+The variable Code is given as a C-style string, it contains the C++ code 
+to be parsed . It has two classes, class A and a templated class B with a member
```
Comment (vgvassilev) [inline docs/tutorials.rst:193]:
```suggestion
to be parsed. It has two classes, class `A` and a templated class `B` with a member 
```

Comment (vgvassilev) [reply docs/tutorials.rst:193]: Please update elsewhere like that.

**Comment by vgvassilev** [comment]
As discussed in the last meeting we had to do some renaming. GitHub decided to close this PR probably because we rewrote our history. Can you reopen the PR against the new version of main?

**Comment by Krishna-13-cyber** [comment]
> As discussed in the last meeting we had to do some renaming. GitHub decided to close this PR probably because we rewrote our history. Can you reopen the PR against the new version of main?

Yes I will reopen this.

**Pull request closed without merging**